railties/lib/rails/generators/primary_file_helpers.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Rails
+  module Generators
+    module PrimaryFileHelpers # :nodoc:
```

Comment:
I'm curious about the naming choice here by the way. Why the word `primary`? Since there's only ever one file that's editable — and thus no `secondary` files — what makes it primary? Especially when this adds an `editor` option and not a `primary_file` option?

How about `Rails::Generators::TemplateEditing` which overrides `template` to support an `edit: true` option?